config/log4j2.yaml

```diff
@@ -0,0 +1,158 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Unspecified loggers and loggers with additivity=true output to server.log and stdout
+# Note that INFO only applies to unspecified loggers, the log level of the child logger is used otherwise
+Configuration:
```

Comment:
Two questions:
1. Have we tested that this log configuration results in the same output as the previous one? In particular, we should avoid anything that requires collecting a stacktrace to log (we made sure of that for the previous configuration).
2. Have we benchmarked the system to make sure there aren't any regressions due to the new logging library? I saw a JIRA/PR saying that log4j2 has a particularly costly `getLogger` implementation.

Reply:
> Have we tested that this log configuration results in the same output as the previous one?

I will use log4j-2 transform tool to check the config later (https://logging.staged.apache.org/log4j/transform/cli.html#log4j-transform-cli)

>  In particular, we should avoid anything that requires collecting a stacktrace to log (we made sure of that for the previous configuration).

Could you please share the commit to me? I trace the history (https://github.com/apache/kafka/commits/3.9/config/log4j.properties) and fails to see the fix about "avoid anything that requires collecting a stacktrace". 

> Have we benchmarked the system to make sure there aren't any regressions due to the new logging library?

According to official docs (https://logging.apache.org/log4j/2.3.x/performance.html), the performance of "logging" has no obvious regression. 

>  I saw a JIRA/PR saying that log4j2 has a particularly costly getLogger implementation.

it seems the story is about `getLogger` - https://issues.apache.org/jira/browse/KAFKA-18046 and https://issues.apache.org/jira/browse/KAFKA-15141 - there is already a PR to fix it #17896 - we can discuss the improvement on it.